src/interpreters/wildDice.ts

```diff
@@ -0,0 +1,319 @@
+import {
+  RollResponse,
+  Rolls,
+  WildType,
+  RollStatus,
+  UnreachableCaseError,
+} from "../utils/lib";
+  
+export const wildDice = (
+  rolls: Rolls,
+  rollType: WildType,
+  zeroD: boolean,
+  cut?: number
+): RollResponse => {
+
+
+  // TODO: cut
+  // TODO: reminder text on 0-dice rolls
+
+    const pool = zeroD ? 0 : rolls.dice.length;
+
+    const doubles: boolean = (() => {
+      // a little space/memory-inefficient, but hopefully not a dealbreaker
+      if (rolls.dice.length !== new Set(rolls.dice).size) {
+        return true;
+      }
+      return false;
+    })();
+
+    if (cut === undefined || cut === null) cut = 0;
+
+    const score = rolls.max;
+    
+    const status: RollStatus =
+      zeroD ? score > 3 // with a pool of 0, roll 1d6, and count triumphs as conflicts
+             ? RollStatus.Mixed
+             : RollStatus.Failure
+
+        : (() => {
+          return score > 5
+          ? RollStatus.Full
+          : score > 3
+          ? RollStatus.Mixed
+          : RollStatus.Failure;
+        })();
+      
+    const title = (() => {
+      switch (rollType) {
+        case WildType.Watch:
+          return (() => {
+            switch (status) {
+              case RollStatus.Full:
+                return `(${rollType}) Peace`;
+              case RollStatus.Mixed:
+                return `(${rollType}) Order`;
+              case RollStatus.Failure:
+                return `(${rollType}) Nature`;
+              default:
+                throw new UnreachableCaseError(status);
+            }
+          })();
+        case WildType.Weather:
+          return (() => {
+            switch (status) {
+              case RollStatus.Full:
+                return `(${rollType}) Clear Skies`;
+              case RollStatus.Mixed:
+                return `(${rollType}) Continuation`;
+              case RollStatus.Failure:
+                return `(${rollType}) A Change for the Worse`;
+              default:
+                throw new UnreachableCaseError(status);
+              }
+            })();
+        default:
+          return (() => {
+            switch (status) {
+
+              case RollStatus.Full:
+                return `(${rollType}) Triumph${
+                  doubles ? " with a twist" : ""
+                }!`;
+
+              case RollStatus.Mixed:
+                return `(${rollType}) Conflict${
+                  doubles ? " with a twist" : ""
+                }!`; 
+                
+              case RollStatus.Failure:
+                return `(${rollType}) Disaster${
+                  doubles ? " with a twist" : ""
+                }!`;
+
+                default:
+                throw new UnreachableCaseError(status);
+              }
+            })();
+ 
+      ;}
+
+    })();
+
+    const description =
+       (() => {
+
+          switch (rollType) {
+            // there is probably a way to avoid this deduplication for the twists text
```

Comment:
```javascript
let returnText = "";
switch (rollType) {
  case WildType.Action:
    returnText += {...more code here...}
```
etc etc.? essentially, start with a blank variable and add pieces on to assemble the proper text for returning. not capital-N Necessary but it might be nice for readability.